smokey/src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -373,6 +384,22 @@ else if(isVerticalArmAtOrAboveHighStop && this.requestedVerticalArmExtension >=
         else if (isVerticalWithinTolerance) {
           this.verticalMotor.set(0.0);
         }
+        // we are nearing puck-zone sometimes too fast
+        else if(currentVerticalExtensionInMeters > lengthVerticalArmExtensionVeryCloseToPucksMeters) {
+          double frogSpellExtensionDistance = 
+            (currentVerticalExtensionInMeters + this.requestedVerticalArmExtension) / 2;
+          verticalPidController.setReference(
+            ArmSubsystem.convertVerticalArmExtensionFromMetersToTicks(frogSpellExtensionDistance),
+            ControlType.kSmartMotion);
+        }
+        // we are nearing bottom stop zone sometimes too fast
+        else if(currentVerticalExtensionInMeters < lengthVerticalArmExtensionVeryCloseToStopMeters) {
+          double frogSpellExtensionDistance = 
```

Comment:
This can be combined with the above block.

Reply:
I think we should leave it as is.